File: docs/protocol/mqtt/js/qubitro_mqtt_publish.js
var mqtt = require('mqtt')

var broker = 'mqtts://broker.qubitro.com'
var deviceId  = 'PASTE_DEVICE_ID'
var deviceToken  = 'PASTE_DEVICE_TOKEN'

var options = {
  keepalive: 60,
  port: 8883,
  username: deviceId,
  password: deviceToken,
  clientId: deviceId,
  clean: false,
  secureProtocol: 'TLSv1_method',
};

var client = mqtt.connect(broker, options);

const data = {
  key: 30,
  key2: 35
}

var sdata = JSON.stringify(data)

client.on('connect', function() {
  console.log("Connected to Qubitro!");
  console.log("Publishing data... ")
    setInterval(
        function(){
            client.publish(deviceId, sdata,function(){
                console.log("Published ", sdata)
            });
        },
    2000)
});

client.on('error', function(err) {
    console.log(err);
});

File: docs/protocol/mqtt/go/qubitro_mqtt_publish.go
package main

import (
	"encoding/json"
	"fmt"
	"time"

	mqtt "github.com/eclipse/paho.mqtt.golang"
)

var deviceID string = "PASTE_DEVICE_ID"
var deviceToken string = "PASTE_DEVICE_TOKEN"
var brokerHost string = "tls://broker.qubitro.com:8883"

var jsonData []byte

//Data is an example payload
type Data struct {
	Temperature float64
	CPUUsage    float64
}

func main() {

	data := &Data{
		Temperature: 40.5,
		CPUUsage:    78,
	}

	jsonData, err := json.Marshal(data)
	if err != nil {
		fmt.Println(err)
	}

	options := mqtt.NewClientOptions().AddBroker(brokerHost)
	options.SetClientID(deviceID)
	options.SetUsername(deviceID)
	options.SetPassword(deviceToken)
	options.AutoReconnect = true

	client := mqtt.NewClient(options)

	if token := client.Connect(); token.Wait() && token.Error() != nil {
		fmt.Println("Error: ", token.Error(), "Visit: https://docs.qubitro.com/client-guides/connect-device/mqtt")
	}

	options.OnConnect = func(c mqtt.Client) {
		fmt.Println("Connected to Qubitro")

		for {
			fmt.Println("Published: ", string(jsonData))
			client.Publish(deviceID, 0, false, jsonData)
			time.Sleep(time.Second * 2)
		}
	}
}
